landingai/common.py

```diff
@@ -75,18 +94,63 @@ class OcrPrediction(Prediction):
     location: List[Tuple[int, int]]
     """A quadrilateral polygon that represents the location of the text. It is a list of four (x, y) coordinates."""
 
+    # Compatibility with BoundingBoxPrediction class
+
+    @property
+    def label(self) -> str:
+        return self.text
+
+    @label.setter
+    def label(self, value: str) -> None:
+        self.text = value
 
-class ObjectDetectionPrediction(ClassificationPrediction):
+    @property
+    def bounding_box(self) -> BoundingBox:
+        x_list = [x for x, _ in self.location]
+        y_list = [y for _, y in self.location]
+        return (
+            min(x_list),
+            min(y_list),
+            max(x_list),
+            max(y_list),
+        )
+
+    @bounding_box.setter
+    def bounding_box(self, value: BoundingBox) -> None:
+        xmin, ymin, xmax, ymax = value
+        self.location = [(xmin, ymin), (xmax, ymin), (xmax, ymax), (xmin, ymax)]
```

Comment:
A bbox is not enough for OcrPrediction as OCR can produce a quadrilateral polygon.
I can help more if you could elaborate more about the model chaining use case and why this change is needed.

Reply:
Yes, I understand that 4 coordinates are not enough to define the OCR text region. 

As said in the previous comment, this is an adapter to make OCRPrediction compatible with the interface above: it will have both the precise 4 points `location` attribute, and the less precise `bounding_box` 4 coordinates (where the crop operation happens).

Reply:
i see. this creates both `location` and `bounding_box` attributes. 
but wouldn't it create confusion for the end user as in why there are two similar attributes?
exposing a `bounding_box` attribute on `OcrPrediction` is also error-prune if user is not aware of the nuance.

how about we just let the users make a conscious choice here? 
let users explicitly convert from quadrilateral polygon to bbox, and keep the current type simple and less error-prone.